include/mbedtls/ecp_internal.h

```diff
@@ -21,61 +21,253 @@
  *
  *  This file is part of mbed TLS (https://tls.mbed.org)
  */
+
+/*
+ * References:
+ *
+ * SEC1 http://www.secg.org/index.php?action=secg,docs_secg
+ * GECC = Guide to Elliptic Curve Cryptography - Hankerson, Menezes, Vanstone
+ * FIPS 186-3 http://csrc.nist.gov/publications/fips/fips186-3/fips_186-3.pdf
+ * RFC 4492 for the related TLS structures and constants
+ *
+ * [Curve25519] http://cr.yp.to/ecdh/curve25519-20060209.pdf
+ *
+ * [2] CORON, Jean-S'ebastien. Resistance against differential power analysis
+ *     for elliptic curve cryptosystems. In : Cryptographic Hardware and
+ *     Embedded Systems. Springer Berlin Heidelberg, 1999. p. 292-302.
+ *     <http://link.springer.com/chapter/10.1007/3-540-48059-5_25>
+ *
+ * [3] HEDABOU, Mustapha, PINEL, Pierre, et B'EN'ETEAU, Lucien. A comb method to
+ *     render ECC resistant against Side Channel Attacks. IACR Cryptology
+ *     ePrint Archive, 2004, vol. 2004, p. 342.
+ *     <http://eprint.iacr.org/2004/342.pdf>
+ */
+
 #ifndef MBEDTLS_ECP_INTERNAL_H
 #define MBEDTLS_ECP_INTERNAL_H
 
 #if defined(MBEDTLS_ECP_INTERNAL_ALT)
 
+/**
+ * \brief           Tell if the cryptographic hardware can handle the group.
+ *
+ * \param grp       The pointer to the group.
+ *
+ * \return          Non-zero if successful.
+ */
 unsigned char mbedtls_internal_ecp_grp_capable( const mbedtls_ecp_group *grp );
 
+/**
+ * \brief           Initialise the crypto hardware accelerator.
+ *
+ *                  If mbedtls_internal_ecp_grp_capable returns true for a
+ *                  group, this function has to be able to initialise the
+ *                  hardware for it.
+ *
+ * \param grp       The pointer to the group the hardware needs to be
+ *                  initialised for.
+ *
+ * \return          0 if successful.
+ */
 int mbedtls_internal_ecp_init( const mbedtls_ecp_group *grp );
 
+/**
+ * \brief           Reset the crypto hardware accelerator to an uninitialised
```

Comment:
Again, the interface is abstract. I think this should simply state that the function 'frees and deallocates the ecp module'.